gaurav/src/components/WithAuth.jsx

```diff
@@ -0,0 +1,92 @@
+import axios from "axios";
+// import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const withAuth = (WrappedComponent) => {
+  const WithAuthComponent = (props) => {
+    // const [auth, setAuth] = useState(false);
+    // const [isLoading, setIsLoading] = useState(true);
+    const navigate = useNavigate();
+    {
+      /*
+            useEffect(() => {
+        checkAuth();
+        }, []);
+
+        const checkAuth = () => {
+        const token = localStorage.getItem("token");
+        if (!token && requireAuth) {
+            navigate("/login");
+            setIsLoading(false);
+            return;
+        }
+        if (token) {
+            try {
+            // Setting Default auth Header
+            axios.defaults.headers.common["Authorization"] = `Bearer ${token}`;
+            setAuth(true);
+
+            if (!requireAuth && window.location.pathname === "/login") {
+                navigate("/");
+            }
+            } catch (error) {
+            console.log("Auth Failed", error);
+            localStorage.removeItem("token");
+            delete axios.defaults.headers.common["Authorization"];
+            if (requireAuth) {
+                navigate("/login");
+            }
+            }
+        }
+        setIsLoading(false);
+        };
+    */
+    }
+    const login = async (data) => {
+      try {
+        const res = await axios.post(
+          "https://api.fr.stg.shipglobal.in/public/api/v1/auth/login",
+          data,
+          {
+            headers: {
+              "Content-Type": "application/json",
+              accept: "application/json",
+            },
+          }
+        );
+        const token = res.data.token;
+        // Storing the token
+        localStorage.setItem("token", token);
+        // Set token in axios headers
+        setAuthToken(token);
+        navigate("/");
```

Comment:
instead of navigating it here, try to use the navigate part in the code where this login function runs with a success.